sdk/eventhub/azure-eventhub/tests/perfstress_tests/_test_base.py

```diff
@@ -3,37 +3,145 @@
+                            batch = await producer.create_batch()
+                            batch.add(EventData(data))
+                    await producer.send_batch(batch)
+                    print(f"Finished loading {events_to_add} events to partition {partition}.")
+
+    async def global_setup(self):
+        await super().global_setup()
+        if self.args.preload:
+            await self._preload_eventhub()
+
+    async def setup(self):
+        """
+        Create the checkpoint store Blob container before starting the Event processors.
+        """
+        if self.args.use_storage_checkpoint:
+            await self.blob_service_client.create_container(self.container_name)
+        await super().setup()
+
+    async def cleanup(self):
+        """
+        Delete the checkpoint store Blob container after closing the Event processors.
+        """
+        await super().cleanup()
+        if self.args.use_storage_checkpoint:
+            await self.blob_service_client.delete_container(self.container_name)
+
+    def stop_events_sync(self) -> None:
+        """
+        Stop the process for receiving events.
+        """
+        self.consumer.close()
+
+    async def stop_events_async(self) -> None:
+        """
+        Stop the process for receiving events.
+        """
+        await self.async_consumer.close()
+
     @staticmethod
     def add_arguments(parser):
-        super(_EventHubTest, _EventHubTest).add_arguments(parser)
+        super(_EventHubProcessorTest, _EventHubProcessorTest).add_arguments(parser)
         parser.add_argument('--event-size', nargs='?', type=int, help='Size of a single event. Defaults to 100 bytes', default=100)
-        parser.add_argument('--num-events', nargs='?', type=int, help='Number of events to send or receive. Defaults to 100', default=100)
+        parser.add_argument('--prefetch-count', nargs='?', type=int, help='Number of events to receive locally per request. Defaults to 300', default=300)
+        parser.add_argument('--load-balancing-strategy', nargs='?', type=str, help="Event Processor load balancing strategy, 'greedy' or 'balanced'. Default is 'greedy'.", default='greedy')
+        parser.add_argument('--checkpoint-interval', nargs='?', type=int, help='Interval between checkpoints (in number of events).  Default is no checkpoints.', default=None)
+        parser.add_argument('--max-wait-time', nargs='?', type=float, help='Maximum time to wait for an event to be received.', default=None)
+        parser.add_argument('--processing-delay', nargs='?', type=int, help='Delay when processing each event (in ms).', default=None)
+        parser.add_argument('--processing-delay-strategy', nargs='?', type=str, help="Whether to 'sleep' or 'spin' during processing delay. Default is 'sleep'.", default='sleep')
+        parser.add_argument('--preload', nargs='?', type=int, help='Ensure the specified number of events are available in each partition. Default is 0.', default=0)
```

Comment:
Do you think it might be more intuitive if the parameter was the total number of events across all partitions, instead of per-partition?  Once you have an estimate of your read ops/sec, you then know the total number of events you need to preload.  It's harder if you have to divide the number of events by number of partitions.